### PR TITLE
Arm: fix test error on Tinker Board (OpenCL on Arm platform)

### PR DESCRIPTION
**Pull request by tomoaki0705**
related #10871 

### This pullrequest changes
  * add vendorID for Arm Platform
  * loosen some test threshold mainly for integer types
  * use relative error for floating points result
  * Log pasted below
    * [opencv_test_video](https://gist.github.com/tomoaki0705/e8553518a98a0bb1437593ee21a3c5f5)
    * [opencv_test_core](https://gist.github.com/tomoaki0705/133af8ccba66204092ab7e949b9fe09c)
    * [opencv_test_imgproc](https://gist.github.com/tomoaki0705/4f047b5502b231ccc3bca4f3829b1ba9)
  * Still, some tests fail on ```opencv_test_imgproc```, but at least those test failure which was caused by too strict threshold has been cleared.
  * I also checked on Geforece GT1060 + Visual Studio 2013
    * this pull request doesn't break any tests, it keeps clean as it was
  * If 3.4.1 is going to be out, I wanted to slip in this PR. (sorry that I couldn't manage to fix all error)

### CODE REVIEW

File: modules/core/src/ocl.cpp [hunk 1]
```diff
@@ -1240,6 +1240,8 @@ struct Device::Impl
             vendorID_ = VENDOR_INTEL;
         else if (vendorName_ == "NVIDIA Corporation")
             vendorID_ = VENDOR_NVIDIA;
+        else if (vendorName_ == "RM")
+            vendorID_ = VENDOR_ARM;
```
Comment (alalek) [inline modules/core/src/ocl.cpp]: Pattern is very generic.

Anyway, these changes are not required for test fixes. Please revert them.

File: modules/imgproc/test/ocl/test_color.cpp [hunk 2]
```diff
@@ -166,10 +166,12 @@ OCL_TEST_P(CvtColor, YUV2BGRA) { performTest(3, 4, CVTCODE(YUV2BGR)); }
 
 // RGB <-> YCrCb
 
-OCL_TEST_P(CvtColor, RGB2YCrCb) { performTest(3, 3, CVTCODE(RGB2YCrCb)); }
-OCL_TEST_P(CvtColor, BGR2YCrCb) { performTest(3, 3, CVTCODE(BGR2YCrCb)); }
-OCL_TEST_P(CvtColor, RGBA2YCrCb) { performTest(4, 3, CVTCODE(RGB2YCrCb)); }
-OCL_TEST_P(CvtColor, BGRA2YCrCb) { performTest(4, 3, CVTCODE(BGR2YCrCb)); }
+#define EPS_EXCEPT_8U(e) (depth == CV_8U ? 1 : e)
```
Comment (alalek) [inline modules/imgproc/test/ocl/test_color.cpp]: `1e-3` makes sense for 32F/64F only. Condition should be changed (with name).

File: modules/video/test/ocl/test_optflowpyrlk.cpp [hunk 3]
```diff
@@ -136,7 +136,7 @@ OCL_TEST_P(PyrLKOpticalFlow, Mat)
                 continue;
             }
 
-            eq = std::abs(cpuErr[i] - err[i]) < 0.01;
+            eq = (std::abs(cpuErr[i] - err[i])/cpuErr[i]) < 0.01;
```
Comment (alalek) [inline modules/video/test/ocl/test_optflowpyrlk.cpp]: Division by zero.

Move "/cpuErr[i]" to rhs of condition:
```
std::abs(cpuErr[i] - err[i]) <= (0.01*std::max(1.0f, cpuErr[i]))
```

**Comment by tomoaki0705** [comment]
Thank you for the review.
Commented places are fixed.

File: modules/core/test/ocl/test_arithm.cpp [hunk 4]
```diff
@@ -867,7 +870,10 @@ OCL_TEST_P(AddWeighted, Mat)
         OCL_OFF(cv::addWeighted(src1_roi, alpha, src2_roi, beta, gama, dst1_roi));
         OCL_ON(cv::addWeighted(usrc1_roi, alpha, usrc2_roi, beta, gama, udst1_roi));
 
-        Near(3e-4);
+    if(dst1_roi.depth() >= CV_32F)
```
Comment (alalek) [inline modules/core/test/ocl/test_arithm.cpp]: Thanks for patch update! Looks good to me :+1:

Please fix `if`/`else` indentation here.

Comment (tomoaki0705) [reply modules/core/test/ocl/test_arithm.cpp]: Ah, right. Thank you for such a fine review !

**Comment by tomoaki0705** [comment]
patch updated

**Pull request merged**